src/crypto.rs

```diff
@@ -59,3 +59,22 @@ pub fn gen_keypair_within_range(range: &RangeInclusive<XorName>) -> Keypair {
         }
     }
 }
+
+#[cfg(test)]
+pub(crate) mod test_utils {
+    use super::*;
+    use ed25519_dalek::SECRET_KEY_LENGTH;
+    use proptest::prelude::*;
+
+    pub(crate) fn arbitrary_keypair() -> impl Strategy<Value = Keypair> {
+        any::<[u8; SECRET_KEY_LENGTH]>().prop_map(|bytes| {
+            // OK to unwrap because `from_bytes` returns error only if the input slice has incorrect
+            // length. But here we only generate arrays of size `SECRET_KEY_LENGTH` which is the
+            // correct one.
+            let secret = SecretKey::from_bytes(&bytes[..]).unwrap();
```

Comment:
Because the idea is to generate a `Keypair` out of arbitrary sequence of bytes, so that we can use it for proptest. `SecretKey::from_bytes` works with any input as long as it has the correct length, but `Keypair::from_bytes` does not. See also https://docs.rs/ed25519-dalek/1.0.1/ed25519_dalek/struct.Keypair.html#warning.